dagger/dagger_test.go

```diff
@@ -0,0 +1,169 @@
+}
+
+type BlockInfo struct {
+	Chain  string `json:"chain"`
+	Height int    `json:"height"`
+	Hash   string `json:"hash"`
+	Time   string `json:"time"`
+}
+
+type Movement struct {
+	Asset    string    `json:"asset"`
+	Contract string    `json:"contract"`
+	AssetID  string    `json:"asset_id"`
+	From     []Address `json:"from"`
+	To       []Address `json:"to"`
+}
+
+type Address struct {
+	Balance   string `json:"balance"`
+	Address   string `json:"address"`
+	AddressID string `json:"address_id"`
+}
+
+type Fee struct {
+	Asset    string `json:"asset"`
+	Contract string `json:"contract"`
+	Balance  string `json:"balance"`
+}
+
+var chain string
+
+func init() {
+	flag.StringVar(&chain, "chain", "", "Specify the blockchain chain to test against")
+}
+
+func getTestTokensFromFaucet(address, amount string) error {
+	url := "http://node-service:10001/chains/ETH/assets/ETH"
+
+	requestBody, err := json.Marshal(map[string]string{
+		"amount":  amount,
+		"address": address,
+	})
+	if err != nil {
+		return fmt.Errorf("error creating request body: %v", err)
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return fmt.Errorf("error creating request: %v", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("error sending request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected status code: %d, body: %s", resp.StatusCode, string(body))
+	}
+
+	fmt.Printf("Response: %s\n", string(body))
+	return nil
+}
+
+func parseTx(data string) (*Tx, error) {
+	var tx Tx
+
+	err := json.Unmarshal([]byte(data), &tx)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	return &tx, nil
+}
+
+func TestChain(t *testing.T) {
+	flag.Parse()
+
+	if chain == "" {
+		t.Fatal("Chain argument is required. Use --chain flag to specify it.")
+	}
+
+	ctx := context.Background()
+
+	address, err := handler.AddressHandler(ctx)
+	if err != nil {
+		t.Fatalf("Error when retriving wallet address: %v", err)
+	}
+	addressStr := string(address)
+
+	amount := "1000000000000000000"
+	err = getTestTokensFromFaucet(addressStr, amount)
+	if err != nil {
+		t.Fatalf("Failed to execute curl command: %v", err)
+	}
+
+	rpcUrl := "http://node-service:10000"
+	args := []string{addressStr, chain, rpcUrl}
+	walletBalance, err := handler.BalanceHandler(ctx, "", args)
+	if err != nil {
+		t.Fatalf("Error when retriving wallet balance: %v", err)
+	}
+
+	assert.Equal(t, walletBalance, amount)
+
+	_, err = handler.TxInputHandler(ctx, "", "", args) // Do we want to check anything from tx input?
+	if err != nil {
+		t.Fatalf("Error when retriving tx input: %v", err)
+	}
+
+	toAddress := "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266"
```

Comment:
this should be derived from a public key (or new private key -> public key -> address).  otherwise it's only going to work on EVM.  It's not valid on Solana or cosmos.